components/brave_ads/core/browser/service/oblivious_http_key_config.h

```diff
@@ -0,0 +1,86 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_ADS_CORE_BROWSER_SERVICE_OBLIVIOUS_HTTP_KEY_CONFIG_H_
+#define BRAVE_COMPONENTS_BRAVE_ADS_CORE_BROWSER_SERVICE_OBLIVIOUS_HTTP_KEY_CONFIG_H_
+
+#include <memory>
+#include <optional>
+#include <string>
+
+#include "base/memory/raw_ptr.h"
+#include "base/memory/scoped_refptr.h"
+#include "base/memory/weak_ptr.h"
+#include "base/timer/wall_clock_timer.h"
+#include "brave/components/brave_ads/core/browser/service/oblivious_http_feature.h"
+#include "url/gurl.h"
+
+class PrefService;
+
+namespace base {
+class TimeDelta;
+}  // namespace base
+
+namespace network {
+class SimpleURLLoader;
+class SharedURLLoaderFactory;
+}  // namespace network
+
+namespace brave_ads {
+
+// Manages the key config required for performing Oblivious HTTP (OHTTP)
+// requests. This includes fetching, caching, validating expiration, and
+// retry/backoff handling.
```

Comment:
Would you mind elaborating on this documentation a bit about the workflow of this class. There are a few things relating to the timer, when it triggers, etc, that would be good to mention.

Reply:
Updated, thanks